x-pack/platform/plugins/shared/inference/server/chat_complete/anonymization/execute_regex_rule.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { RegexAnonymizationRule } from '@kbn/inference-common';
+import { AnonymizationState } from './types';
+import { runRegexTask } from './regex_worker';
+
+/**
+ * Executes a regex anonymization rule, by iterating over the matches,
+ * and replacing each occurrence with a masked value.
+ */
+export async function executeRegexRule({
+  state,
+  rule,
+}: {
+  state: AnonymizationState;
+  rule: RegexAnonymizationRule;
+}): Promise<AnonymizationState> {
+  const { nextRecords, anonymizations } = await runRegexTask({
```

Comment:
any notion of the overhead of running each regex rule in a worker thread separately vs batching them?

Reply:
I think it should be fine because these run sequentially if we set idleTimeout > 0, we'll have warm workers

Reply:
I tested with large paragraphs of text around 20 regexes and I couldn't see a difference whether or not `idleTimeout` was set - they both happen very quickly.  I think we'd need a very large number of rules to notice.  I've went ahead and set `idleTimeout` to 30000.